File: app/src/main/java/com/worldventures/dreamtrips/wallet/domain/converter/SmartCardLocationToWalletLocationConverter.java
package com.worldventures.dreamtrips.wallet.domain.converter;

import com.worldventures.dreamtrips.api.smart_card.location.model.SmartCardLocation;
import com.worldventures.dreamtrips.modules.mapping.converter.Converter;
import com.worldventures.dreamtrips.wallet.domain.entity.lostcard.ImmutableWalletLocation;
import com.worldventures.dreamtrips.wallet.domain.entity.lostcard.WalletCoordinates;
import com.worldventures.dreamtrips.wallet.domain.entity.lostcard.WalletLocation;
import com.worldventures.dreamtrips.wallet.domain.entity.lostcard.WalletLocationType;

import io.techery.mappery.MapperyContext;


public class SmartCardLocationToWalletLocationConverter implements Converter<SmartCardLocation, WalletLocation> {
   @Override
   public Class<SmartCardLocation> sourceClass() {
      return SmartCardLocation.class;
   }

   @Override
   public Class<WalletLocation> targetClass() {
      return WalletLocation.class;
   }

   @Override
   public WalletLocation convert(MapperyContext mapperyContext, SmartCardLocation smartCardLocation) {
      return ImmutableWalletLocation.builder()
            .coordinates(mapperyContext.convert(smartCardLocation.coordinates(), WalletCoordinates.class))
            .createdAt(smartCardLocation.createdAt())
            .type(mapperyContext.convert(smartCardLocation.type(), WalletLocationType.class))
            .build();
   }
}

File: wallet/src/main/java/com/worldventures/wallet/domain/converter/SmartCardLocationToWalletLocationConverter.kt
package com.worldventures.wallet.domain.converter

import com.worldventures.core.converter.Converter
import com.worldventures.dreamtrips.api.smart_card.location.model.SmartCardLocation
import com.worldventures.wallet.domain.entity.lostcard.WalletCoordinates
import com.worldventures.wallet.domain.entity.lostcard.WalletLocation
import com.worldventures.wallet.domain.entity.lostcard.WalletLocationType
import io.techery.mappery.MapperyContext

class SmartCardLocationToWalletLocationConverter : Converter<SmartCardLocation, WalletLocation> {

   override fun sourceClass(): Class<SmartCardLocation> = SmartCardLocation::class.java

   override fun targetClass(): Class<WalletLocation> = WalletLocation::class.java

   override fun convert(context: MapperyContext, source: SmartCardLocation): WalletLocation {
      return WalletLocation(
            coordinates = if (source.coordinates() != null) context.convert(source.coordinates()!!, WalletCoordinates::class.java) else WalletCoordinates(0.0, 0.0), // todo
            createdAt = source.createdAt(),
            type = context.convert(source.type(), WalletLocationType::class.java))
   }
}
